homeassistant/components/ring/coordinator.py

```diff
@@ -91,19 +104,105 @@ async def _async_update_data(self) -> RingDevices:
         return devices
 
 
-class RingNotificationsCoordinator(DataUpdateCoordinator[None]):
+class RingListenCoordinator(BaseDataUpdateCoordinatorProtocol):
     """Global notifications coordinator."""
 
-    def __init__(self, hass: HomeAssistant, ring_api: Ring) -> None:
+    config_entry: config_entries.ConfigEntry
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        ring_api: Ring,
+        listen_credentials: dict[str, Any] | None,
+        listen_credentials_updater: Callable[[dict[str, Any]], None],
+    ) -> None:
         """Initialize my coordinator."""
-        super().__init__(
-            hass,
-            logger=_LOGGER,
-            name="active dings",
-            update_interval=NOTIFICATIONS_SCAN_INTERVAL,
-        )
+        self.hass = hass
+        self.logger = _LOGGER
         self.ring_api: Ring = ring_api
-
-    async def _async_update_data(self) -> None:
-        """Fetch data from API endpoint."""
-        await _call_api(self.hass, self.ring_api.async_update_dings)
+        self.event_listener = RingEventListener(
+            ring_api, listen_credentials, listen_credentials_updater
+        )
+        self._listeners: dict[CALLBACK_TYPE, tuple[CALLBACK_TYPE, object | None]] = {}
+        self._listen_callback_id: int | None = None
+
+        config_entry = config_entries.current_entry.get()
+        if TYPE_CHECKING:
+            assert config_entry
+        self.config_entry = config_entry
```

Comment:
coordinator already sets this

https://github.com/home-assistant/core/blob/ab29718a455ed6c625a0459400c816f293e695b8/homeassistant/helpers/update_coordinator.py#L87

Reply:
I just realized you are only using the protocol.  

Why can't we use the coordinator anymore?

Reply:
I'm guessing you don't want to poll....

Reply:
Hah, this was also my remark. But I guess it's also possible to use a normal coordinator while being push based